### PR TITLE
#16: Player-specific logic

### PR DESCRIPTION
**Pull request by bbqbaron**
Whoo, was this one a doozy! I am perfectly happy to change any implicit or explicit design call in here. It didn't take as long as the list of changes might look; mostly a bunch of moving statements around.

**High Level** Ensure that the app understands that some things affect specific players and others don't. Make sure the app doesn't _completely_ lose its mind while two people are connected.

* Add `playerId` and `players` keys to state.
* New assumption: actions dispatched with a falsey `playerId` are implicitly "mine".
* Add a `PICK_PLAYER` action.
* Add a _very_ crude component at load that prompts you to pick player 1 or 2. This doesn't even qualify for the word "lobby" yet :)
* Make some of the `didMount` side effects idempotent
* Refactor `gameState` reducer. Create a separate wrapper that reduces particular player states, as opposed to the central state. This is a _little_ weird in that, for example, it has both reducers checking whether a player can afford a card. They may be _too_ decoupled, sorry :)
* Update tests
* Update component-side references to `gameState` to target specific players.
* Add a `prettier` formatter command for fun

Aside: I tend to write extremely expression-oriented/functional code; I reach for `map`, `filter`, `reduce` etc as a first instinct. Is this ok? I don't want to ruin your flow if you find flow like this hard to parse! It helps me break down an answer into sub-answers, like solving a math problem.

### CODE REVIEW

**Comment by jhmullen** [comment]
Re: your aside, I don't mind at all -- one of my goals in this project is to explore some different programming practices, so I'm glad to follow along as I can!

I checked out this branch and found a few small things that I'll ask about in the review section.

**Comment by bbqbaron** [comment]
They are definitely not intentional 😬  and I should have tested more thoroughly. I'll take a look soon; fixes incoming, having actually run through more play scenarios.

**Comment by bbqbaron** [comment]
Holy shit, that was horrifically broken. I must have made some "trivial" changes after at least smoke-testing during dev; it's the only excuse I can make. Ok, that and working in langs like Ocaml, whose compiler backstops have made my brain soft and worthless. Weighing quitting programming forever.
:rage4: :rage4: :rage4: :rage4: :rage4: :rage4: :rage4: :rage4: :rage4: :rage4: :rage4: 

Ok, breathe. Meantime, fixes have arrived for the most obvious things, including bonus penance fix for an older bug. The obvious stuff seems to be working so far. Many more reducer tests needed, for sure.

I might be missing something; is it possible that `PLAY_CARD` was already not refilling the player's hand on master? Naturally should fix it; just want to understand its context.

The big remaining item that I already know of is one you raised on #16--the fact that starter cards are no longer unique in-game. Yes, I think we should definitely make them unique _or_ remove expectations that card `id` refers to a single unique instance of the card. Down the line, I can see needing ids for both the card's rules data and the "physical" instance, but one step at a time--I don't think adding some kind of playerId suffix is _that_ bad a hack for now. 👍 ?

**Comment by jhmullen** [comment]
Thanks for sending an updated branch!  I should have some time to review this later today.  Hope you had a good holiday by the way 🎄

**Comment by jhmullen** [comment]
No need to quit programming haha!  Your output and test discipline and stuff have been inspiring to say the least, at least for my newer React brain ;)  I appreciate the fixes.

I still see some small weirdness with player 2, which seems to be related to our playerID suffix.  I'm going to merge this in and have a look.
